commands/rm.go

```diff
@@ -43,30 +44,41 @@ func runRm(dockerCli command.Cli, in rmOptions) error {
 		return rmAllInactive(ctx, txn, dockerCli, in)
 	}
 
+	var ng *store.NodeGroup
 	if in.builder != "" {
-		ng, err := storeutil.GetNodeGroup(txn, dockerCli, in.builder)
+		ng, err = storeutil.GetNodeGroup(txn, dockerCli, in.builder)
 		if err != nil {
 			return err
 		}
-		err1 := rm(ctx, dockerCli, in, ng)
-		if err := txn.Remove(ng.Name); err != nil {
+	} else {
+		ng, err = storeutil.GetCurrentInstance(txn, dockerCli)
+		if err != nil {
 			return err
 		}
-		return err1
+	}
+	if ng == nil {
+		return nil
 	}
 
-	ng, err := storeutil.GetCurrentInstance(txn, dockerCli)
+	ctxbuilders, err := dockerCli.ContextStore().List()
 	if err != nil {
 		return err
 	}
-	if ng != nil {
-		err1 := rm(ctx, dockerCli, in, ng)
-		if err := txn.Remove(ng.Name); err != nil {
-			return err
+	for _, cb := range ctxbuilders {
+		if ng.Name == "default" && len(ng.Nodes) == 1 && ng.Nodes[0].Endpoint == cb.Name {
```

Comment:
Not sure why there is a "default" name check in here. Shouldn't it check that `ng.Driver == "docker"` ? What does the "default" mean in here?

Didn't we already have an error suggesting to use `docker context` somewhere or am I mixing it up with some other case.

Reply:
`docker-container` is not the default ever, it always requires an instance to be created and therefore can always be removed.

Reply:
as discussed I switched to the cond on the driver name.